dataikuapi/dssclient.py

```diff
@@ -626,8 +649,73 @@ def create_global_api_key(self, label=None, description=None, admin=False):
             raise Exception('API key creation failed : %s' % (json.dumps(resp.get('messages', {}).get('messages', {}))))
         if not resp.get('id', False):
             raise Exception('API key creation returned no key')
-        key = resp.get('key', '')
-        return DSSGlobalApiKey(self, key)
+
+        if as_type == 'object':
+            return DSSGlobalApiKey(self, resp.get('id', ''))
+        elif as_type == 'dict':
+            return resp
+        else:
+            raise ValueError("Unknown as_type")
+    ########################################################
+    # Personal API Keys
+    ########################################################
+
+    def list_personal_api_keys(self, as_type='dict'):
+        """
+        List all personal API keys:
+            - not admin: only the keys belonging to the owner of the key
+            - admin: all the personal keys
+
+        :param str as_type: 'objects' or 'dict'
+
+        :returns: All personal API keys
+        """
+
+        resp = self._perform_json(
+            "GET", "/personal-api-keys/")
+        if as_type == 'objects':
+            return [DSSPersonalApiKey(self, item.id) for item in resp]
+        elif as_type == 'dict':
+            return resp
+        else:
+            raise ValueError("Unknown as_type")
+
+    def get_personal_api_key(self, id_):
+        """
+        Get a specific API key from the user doing the request
+
+        :param str id_: the id the desired API key
+
+        :returns: A :class:`dataikuapi.dss.admin.DSSPersonalApiKey` API key
+        """
+        return DSSPersonalApiKey(self, id_)
+
+    def create_personal_api_key(self, label="", description="", as_type='dict', on_behalf=""):
```

Comment:
I think that we have many issues with the `on_behalf` parameter.

Currently :
1. Anyone can create a personal key on behalf of anyone and most problematically on behalf of admin.
2. Kiss cool effect: if you ask for a `dict` as return type of the creation command, you are now the happy owner of a personal api key of `admin`.
3. If you don't specify `on_behalf` (it's currently optional and defaulted to an empty string, that the backend interprets as not null which force the `onBehalf` to be empty) you have an unusable key that only an admin clean up. 

All of this comes from the backend code but I prefer to have it here also to double check that it's working in the end.